### PR TITLE
Add support for manage_repo

### PR DESCRIPTION
**Pull request by eabecho**
Added parameters:
- manage_repo
- repo_base_url
- esx_version
- gpgkey_url
- proxy_host
- proxy_port
- prefer_open_vm_tools

Removed parameters:
- manage_repo_package
- repo_package_name
- repo_package_ensure
- manage_tools_kmod_package
- tools_kmod_package_name
- tools_kmod_package_ensure

### CODE REVIEW

File: Modulefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name    'puppet-module-vmware'
-version '0.2.0'
+version '0.3.0'
```
Comment (emahags) [inline Modulefile]: This is a major change, bump it to 1.0.0

**Comment by emahags** [comment]
I'll run some tests in our environment to make sure there are no unexpected changes. Meanwhile you should make sure the travis checks go through.

**Comment by anders-larsson** [comment]
@eabecho @emahags is sadly busy but I promised him to look at your modifications.
Your changes seem to work nicely on RHEL64 (with standard settings). However on SLE11.2 I'm getting GPG error using the default settings. For some reason it does not seem like the GPG key got installed with zypprepo.

Contents of vmware-osps.repo:
````
[vmware-osps]
baseurl=http://packages.vmware.com/tools/esx/5.5/sles11.2/x86_64
path=/
enabled=1
gpgcheck=1
gpgkey=http://packages.vmware.com/tools/keys/VMWARE-PACKAGING-GPG-RSA-KEY.pub
autorefresh=0
type=yum
````

What we feel is missing is a way to get the gpg-key installed even though you're not using manage_repo = true. It's not always possible (or preferable) to download packages directly from the Internet. A solution might perhaps be to allow the user to install a custom package containing the gpg key. This is the solution currently used in the module.

Regarding the issue you're seeing with travis it seems to be the same issue I had (and Johan Wennerberg). You should be able to solve it by this addition: https://github.com/jwennerberg/puppet-module-pam/commit/55abd50c8b84eb628660c9a07b2015f718a16eed

**Comment by anders-larsson** [comment]
Hm. A few (maybe 3) puppet runs later it worked correctly on SLE11.2. Not sure what might cause it.

**Comment by eabecho** [comment]
I'm assuming that there's a mirror of the VMware packages repo.  Have you tried creating a local mirror?

I used the following script:
```
#!/bin/sh                                                                       

export http_proxy=http://www-proxy.ericsson.se:8080/

ESXVERSIONS="5.5u2"
FLAGS='--mirror --no-parent'
RFLAGS="-R *\??=?,index.html"
# not working                                                                   
XFLAGS="-X repos,rhel3,rhel4"

wget $FLAGS $RFLAGS $XFLAGS http://packages.vmware.com/tools/keys/
for esx in $ESXVERSIONS; do
    wget $FLAGS $RFLAGS $XFLAGS http://packages.vmware.com/tools/esx/$esx/
done
```

**Comment by eabecho** [comment]
Thanks for the tip re: Travis.  That fixed the problem.

**Comment by anders-larsson** [comment]
We've created a custom channel in the Red Hat Satellite called <Channel Name> Custom which contains all non-standard packages that we want distributed to our systems. This gives us control of the packages available to our systems by default. I'm certain that a similar approach is possible with SuSE Manager.

**Comment by eabecho** [comment]
I was trying to get away from needing to have anything to do with custom packages given how many packages would be needed to be updated over time (depending upon the version esx).

It seems that if one were to manage one's own repo, one should take care of any keys as part of it (i.e. it doesn't seem appropriate to have this module take care of the gpgkey independent of a repo).  I know Red Hat Satellite has the functionality to associate the gpg key with a channel.  I'm not familiar with SuSE Manager.

**Comment by eabecho** [comment]
Any updates?

**Comment by emahags** [comment]
@eabecho: Just two more comments before I merge this.

- You have introduced some errors in puppet-lint, please fix them (I saw some others as well, but they existed before. I'll fix those after this merge):
```
$ puppet-lint --no-80chars-check .
./manifests/init.pp - WARNING: class included by relative name on line 144
./manifests/init.pp - WARNING: class included by relative name on line 172
./manifests/init.pp - WARNING: class included by relative name on line 175
```
- I feel that the commit message doesn't reflect what has changed anymore, since this is pretty much a complete rework of the module.

Once this is fixed I'll merge the commit.

**Comment by eabecho** [comment]
I'm not getting those warnings in my puppet-lint.

eabecho@eussjlts0102$ /opt/puppet/bin/puppet-lint --no-80chars-check .
./manifests/init.pp - WARNING: quoted boolean value found on line 110
./manifests/init.pp - WARNING: quoted boolean value found on line 112
./manifests/init.pp - ERROR: vmware not in autoload module layout on line 5

I updated the commit message.  What would you suggest as the commit message?

**Comment by emahags** [comment]
Ah, I guess it was one I installed from http://www.camptocamp.com/actualite/getting-code-ready-puppet-4/. You could do that as well or just leave it and I'll get them when I fix the other warnings.

For the commit message I guess something like "Rework module to better support repositories" would be ok.

**Comment by Phil-Friderici** [comment]
@emahags Thanks for the nice link to puppet-lint v4 plugins :)

**Comment by eabecho** [comment]
Updated commit message per recommendation.

**Comment by emahags** [comment]
@eabecho It looks like there is something weird going on with the commits. Did you amend your changes to my commit?

**Comment by eabecho** [comment]
Sorry for the messed up logs.  I redid the whole checkout.

**Comment by emahags** [comment]
@eabecho Thanks for the huge contribution!

**Pull request merged**